src/tests/multihost/ad/conftest.py

```diff
@@ -132,6 +132,35 @@ def remove_ad_user_group():
     return (ad_user, ad_group)
 
 
+@pytest.fixture(scope="function")
+def create_nested_group(session_multihost, create_aduser_group, request):
+    """ create nested AD group
+        l1_grp is a top-level group
+        l2_grp is a level-2 group and is a member of l1_grp group
+        ad_user is a user-member of a 2nd-level l2_grp group
+    """
+    ad_user = f'testuser-{random.randint(999, 999999)}'
```

Comment:
If You use the same random id for all the users and groups from the fixture, you can easily return it from the fixture like done in create_aduser_group.
```
run_id = random.randint(999, 999999)}
ad_user = f'testuser-{run_id}'
ad_group = f'testgrp-{run_id}'
l1_grp = f'testgrp-l1-{run_id}'
l2_grp = f'testgrp-l2-{run_id}'
...
return run_id
```

In the test it will be:
```
run_id = create_aduser_group
ad_user = f'testuser-{run_id}'
ad_group = f'testgrp-{run_id}'
l1_grp = f'testgrp-l1-{run_id}'
l2_grp = f'testgrp-l2-{run_id}'
```

Reply:
Yes, this is good suggestion. Modifying the fixture. Here, the fixture now would only return 'run_id' and caller testcase will then define required aduser, adgroup variable. The testcase here needs aduser and top-level nested group to be defined.